### ISSUE TITLE
Turkish characters are not supported.

### ISSUE THREAD

**Issue by HamzaYslmn** [jorisschellekens/borb#173]
**Is your feature request related to a problem? Please describe.**
I would like to prepare an invoice, but the ones like "I Ğ ğ ı ₺ ş" and others are not supported. 

**Describe the solution you'd like**
custom ttf support may come idk. or at least a times new roman that supports Turkish can be added to afm files. I don't know how to solve it.

**Additional context**

<img width="392" alt="image" src="https://github.com/jorisschellekens/borb/assets/78810304/9465101b-91c6-4a77-a6bc-7179d3e27fba">
<img width="199" alt="image" src="https://github.com/jorisschellekens/borb/assets/78810304/22ca7a76-657a-4e87-916b-bf475c98bb08">

AssertionError: Font Times-Roman can not represent 'ı'
AssertionError: Font Times-Roman can not represent 'İ'
AssertionError: Font Times-Roman can not represent '₺'

**Comment by HamzaYslmn** [comment]
I don't use it commercially, I'm a university student and I wanted a pdf creator that could give a good result. It was tiring to use FPDF.

thanks again for this beautiful library, the detailed documentation is very useful

**Comment by jorisschellekens** [comment]
You have not yet shared a piece of **code** that actually produces the bug.
Also, don't share code/data as screenshots. If you're sharing code/data as screenshots, you are forcing the reader (me) to re-type everything. Make it as easy as possible for me to help you.

Kind regards,
Joris Schellekens

**Comment by HamzaYslmn** [comment]
There is no error in the code to make:

Fatura Numarası > Fatura Numarasi
Cari İsim > Cari Isim
₺ > TL 

I am also very new to coding, I have no idea where it is generating errors. Anyway thank you for your help <3

**Comment by HamzaYslmn** [comment]
from borb.pdf import Document
from borb.pdf import Page
from borb.pdf import SingleColumnLayout
from borb.pdf import PageLayout
from decimal import Decimal
from borb.pdf import Image
from borb.pdf import FixedColumnWidthTable
from borb.pdf import Paragraph
from borb.pdf import Alignment
from borb.pdf import HexColor, X11Color
from borb.pdf import TableCell
from borb.pdf import PDF

def generate_invoice_pdf(doc, invoice_data):
    page = Page()
    doc.add_page(page)

    page_layout = SingleColumnLayout(page)
    page_layout.vertical_margin = page.get_page_info().get_height() * Decimal(0.02)

    image_url = "https://s3.stackabuse.com/media/articles/creating-an-invoice-in-python-with-ptext-1.png"
    image = Image(image_url, width=Decimal(64), height=Decimal(64))
    page_layout.add(image)

    table = FixedColumnWidthTable(number_of_rows=8, number_of_columns=2)
    for label, value in [
        ("Fatura Numarasi:", invoice_data["invoice_number"]),
        ("Tarih:", invoice_data["date"]),
        ("Cari isim:", invoice_data["name"]),
        ("Brüt Tutar:", invoice_data["gross_amount"]),
        ("KDV:", invoice_data["kdv"]),
        ("ÖTV:", invoice_data["otv"]),
        ("Toplam Tutar:", invoice_data["total"]),
        ("Para birimi:", invoice_data["currency"]),
    ]:
        table.add(TableCell(Paragraph(label, font="Times-roman")))
        table.add(TableCell(Paragraph(str(value), font="Times-roman")))
    table.set_padding_on_all_cells(Decimal(2), Decimal(2), Decimal(2), Decimal(2))
    table.no_borders()
    page_layout.add(table)

    return doc


def generate_invoice_pdf_file(pdf_file_path, invoice_data):
    pdf = Document()
    generate_invoice_pdf(pdf, invoice_data)

    with open(pdf_file_path, "wb") as out_file_handle:
        PDF.dumps(out_file_handle, pdf)


if __name__ == "__main__":
    sample_invoice_data = {
        "invoice_number": "12345",
        "date": "2023-08-22",
        "name": "Sample Customer",
        "gross_amount": "$1000.00",
        "kdv": "$150.00",
        "otv": "$50.00",
        "total": "$1200.00",
        "currency": "₺",
    }

    generate_invoice_pdf_file("output_borb.pdf", sample_invoice_data)

**Comment by jorisschellekens** [comment]
Ok, that's almost useful.
But it would be great if the example was **minimal**.

This is the smallest example that reproduces the error:

```python
from borb.pdf import Document, PDF
from borb.pdf import Page
from borb.pdf import SingleColumnLayout
from borb.pdf import PageLayout
from borb.pdf import Paragraph


def main():

    d: Document = Document()

    p: Page = Page()
    d.add_page(p)

    l: PageLayout = SingleColumnLayout(p)
    l.add(Paragraph("₺"))

    with open("output.pdf", "wb") as fh:
        PDF.dumps(fh, d)

if __name__ == "__main__":
    main()
```

You will have to use your own font, with Turkish characters if you want to display Turkish.
There are 14 standard fonts defined in the PDF specification (which is available in the repository).
The encoding for those fonts is also defined in the spec. 
None of those fonts is meant to display Turkish. 

I am not going to change them in my repository because:
- That would mean that `borb` needs to break the standard
- That would mean that other people might ask me to include even more characters (why not Chinese? Korean? Japanese? Russian? etc) Before you know it, I would have to include fonts that are very large (because they have definitions for all possible characters in unicode)

In short, the library is right.
Use a font that is capable of displaying the character.

Kind regards,
Joris Schellekens
